src/pybamm/expression_tree/interpolant.py

```diff
@@ -107,6 +107,9 @@ def __init__(
                     "len(x1) should equal y=shape[0], "
                     f"but x1.shape={x1.shape} and y.shape={y.shape}"
                 )
+            # if 1d then x should be monotonically increasing
```

Comment:
If the comparison is about monotonically increasing values, then I suppose the comparison should be `>` and not `>=` (see below?)

Reply:
Ah, I perceived `monotonically` as strictly increasing, so `>=` should be good here. Resolving!

Reply:
Ah, I perceived "monotonically" as "strictly" increasing, so `>=` should be good here. Resolving!